### PR TITLE
Make the target works for exe.

### LINKED ISSUE #25320: SGEN need support .exe
It only works for .dll not works for .exe. This is because in the target, we always load <AssemblyName>.dll. In full framework, the output is <AssemblyName>.exe.

### PR DESCRIPTION
**Pull request by huanwu**
Fix #25320 
@shmao @zhenlan @mconnew

### CODE REVIEW

File: src/Microsoft.XmlSerializer.Generator/pkg/build/Microsoft.XmlSerializer.Generator.targets [hunk 1]
```diff
@@ -4,14 +4,19 @@
     <_SerializerDllIntermediateFolder>$(IntermediateOutputPath)$(_SerializationAssemblyName).dll</_SerializerDllIntermediateFolder>
     <_SerializerPdbIntermediateFolder>$(IntermediateOutputPath)$(_SerializationAssemblyName).pdb</_SerializerPdbIntermediateFolder>
     <_SerializerCsIntermediateFolder>$(IntermediateOutputPath)$(_SerializationAssemblyName).cs</_SerializerCsIntermediateFolder>
-    <_SGenWarningText>SGEN : warning SGEN1: Fail to generate the serializer for $(AssemblyName).dll. Please follow the instructions at https://go.microsoft.com/fwlink/?linkid=858594 and try again.</_SGenWarningText>
   </PropertyGroup>
   <Target Name="GenerateSerializationAssembly" AfterTargets="Build">
+    <PropertyGroup>
+      <_fullAssemblyName Condition="Exists('$(IntermediateOutputPath)$(AssemblyName).dll') == 'true'">$(AssemblyName).dll</_fullAssemblyName>
```
Comment (shmao) [inline src/Microsoft.XmlSerializer.Generator/pkg/build/Microsoft.XmlSerializer.Generator.targets]: _fullAssemblyName -> _OutputAssemblyFullName

Can we get the file extension from output type?

Comment (huanwu) [reply src/Microsoft.XmlSerializer.Generator/pkg/build/Microsoft.XmlSerializer.Generator.targets]: The outputtype is always exe although the assembly is end with .dll.

Comment (shmao) [reply src/Microsoft.XmlSerializer.Generator/pkg/build/Microsoft.XmlSerializer.Generator.targets]: I see. How about TargetExt?

**Comment by zhenlan** [comment]
Does this work for asp.net core apps (when target framework is netcoreapp2.0 and net461)?

**Comment by huanwu** [comment]
@zhenlan It works when set TargetFramework to net461 or netcoreapp2.0.

**Comment by stephentoub** [comment]
@huanwu, what's the status of this PR? Is there still work to be done, or can it be squashed and merged?

**Pull request merged**